File: Merge_Sorted_Array.cpp
class Solution {
public:
    /**
     * @param A: sorted integer array A which has m elements, 
     *           but size of A is m+n
     * @param B: sorted integer array B which has n elements
     * @return: void
     */
    void mergeSortedArray(int A[], int m, int B[], int n) {
        // write your code here
        int length = m + n - 1;
        --m; --n;
        while ( m > -1 && n > -1){
            if (A[m] > B[n]){
                A[length] = A[m];
                --m;
            }else{
                A[length] = B[n];
                --n;
            }
            --length;
        }
        while (n > -1){
            A[length] = B[n];
            --length;
            --n;
        }
    }
};

File: Merge Sorted Array.py
class Solution:
    """
    @param A: sorted integer array A which has m elements, 
              but size of A is m+n
    @param B: sorted integer array B which has n elements
    @return: void
    """
    def mergeSortedArray(self, A, m, B, n):
        # write your code here
        
        i = m - 1
        j = n - 1
        index = m + n - 1
        while i >= 0 and j >= 0:
            if A[i] > B[j]:
                A[index] = A[i]
                i -= 1
            else:
                A[index] = B[j]
                j -= 1
            index -= 1
        
        if j >= 0:
            A[:j + 1] = B[:j + 1]
        
        